plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/function/tablechanges/DeltaLakeTableChangesFunction.java

```diff
@@ -0,0 +1,190 @@
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.function.SchemaFunctionName;
+import io.trino.spi.ptf.AbstractConnectorTableFunction;
+import io.trino.spi.ptf.Argument;
+import io.trino.spi.ptf.Descriptor;
+import io.trino.spi.ptf.ScalarArgument;
+import io.trino.spi.ptf.ScalarArgumentSpecification;
+import io.trino.spi.ptf.TableFunctionAnalysis;
+import io.trino.spi.type.TypeManager;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.deltalake.DeltaLakeMetadata.toColumnHandle;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.extractSchema;
+import static io.trino.spi.StandardErrorCode.NOT_FOUND;
+import static io.trino.spi.ptf.ReturnTypeSpecification.GenericTable.GENERIC_TABLE;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MILLIS;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class DeltaLakeTableChangesFunction
+        extends AbstractConnectorTableFunction
+{
+    public static final String TABLE_NAME_ARGUMENT = "TABLE_NAME";
+    public static final String START_VERSION_ARGUMENT = "START_VERSION";
+    public static final String END_VERSION_ARGUMENT = "END_VERSION";
+    public static final String SCHEMA_NAME = "system";
+    public static final String NAME = "table_changes";
+    public static final SchemaFunctionName TABLE_CHANGES_NAME = new SchemaFunctionName(SCHEMA_NAME, NAME);
+    public static final String CHANGE_TYPE_COLUMN_NAME = "_change_type";
+    public static final String COMMIT_VERSION_COLUMN_NAME = "_commit_version";
+    public static final String COMMIT_TIMESTAMP_COLUMN_NAME = "_commit_timestamp";
+
+    private final HiveMetastoreFactory hiveMetastoreFactory;
+    private final TransactionLogAccess transactionLogAccess;
+    private final TypeManager typeManager;
+    private final CachingExtendedStatisticsAccess statisticsAccess;
+    private final TrinoFileSystemFactory fileSystemFactory;
+
+    public DeltaLakeTableChangesFunction(
+            HiveMetastoreFactory hiveMetastoreFactory,
+            TransactionLogAccess transactionLogAccess,
+            TypeManager typeManager,
+            CachingExtendedStatisticsAccess statisticsAccess,
+            TrinoFileSystemFactory fileSystemFactory)
+    {
+        super(
+                SCHEMA_NAME,
+                NAME,
+                ImmutableList.of(
+                        ScalarArgumentSpecification.builder().name(TABLE_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(START_VERSION_ARGUMENT).type(BIGINT).defaultValue(-1L).build(),
```

Comment:
Why `-1` ?
There is no such version to my knowledge in a delta lake table.

Reply:
Default `null` isn't great either, but probably more readable that `-1`

Reply:
So that is the best hack I could think of to be able to read all changes if no argument is provided and to have startVersion being exclusive. 
If this -1 is used it will be always increased to 0 in `io.trino.plugin.deltalake.function.tablechanges.DeltaLakeTableChangesFunction#analyze`

Reply:
ok I will make it work with null